core/test/scripts/Voting.js

```diff
@@ -195,6 +195,33 @@ contract("scripts/Voting.js", function(accounts) {
     assert.equal((await voting.getPrice(identifier, time)).toString(), web3.utils.toWei("9155.05"));
   });
 
+  it("Notification on crash", async function() {
+    // Force a Solidity revert by returning an invalid hash in the soliditySha3 function.
+    const temp = web3.utils.soliditySha3;
+    web3.utils.soliditySha3 = () => {
+      return "0x0";
+    };
+
+    const identifier = web3.utils.utf8ToHex("Custom Index (1)");
+    const time = "1560762000";
+
+    // Request an Oracle price.
+    await voting.addSupportedIdentifier(identifier);
+    await voting.requestPrice(identifier, time);
+
+    // Run an iteration, which should crash.
+    const notifier = new MockNotifier();
+    const votingSystem = new VotingScript.VotingSystem(voting, voter, [notifier]);
+    await moveToNextRound(voting);
+    await votingSystem.runIteration();
+
+    // A notification email should be sent.
+    assert.equal(notifier.notificationsSent, 1);
```

Comment:
Just so I understand, would this test still succeed even if `runIteration` didn't crash? (I think that's fine, just wanted to double check my understanding)

Reply:
Yep. What do you think of this version? Feels somewhat safer to mock on an object we control.

Reply:
Agreed, this is way better!